subprojects/core/src/main/java/org/gradle/api/internal/provider/DefaultProviderFactory.java

```diff
@@ -69,9 +67,9 @@ public DefaultProviderFactory(FileOperations fileOperations) {
             ((PropertyState<Double>) propertyState).set(Double.valueOf(0));
         } else if (clazz == Character.class) {
             ((PropertyState<Character>) propertyState).set(new Character('\0'));
-        } else if (clazz == FileCollection.class || clazz == ConfigurableFileCollection.class) {
+        } else if (clazz == ConfigurableFileCollection.class) {
```

Comment:
I get it now. Thanks for clarifying. I removed it from the factory and adapted the tests/samples.

I agree that we should work on the FileCollection is-a Provider as the next story.